sdk/servicebus/service-bus/test/atomManagement.spec.ts

```diff
@@ -47,364 +48,444 @@ const alwaysBeDeletedSubscription = "alwaysbedeletedsubscription";
+    });
 
-          case EntityType.SUBSCRIPTION:
-            response = await listEntities(entityType, alwaysBeExistingTopic);
-            break;
+    it(`Lists on existing entities for type ${testCase.entityType} with top 1 returns the first entity`, async () => {
+      let response;
 
-          case EntityType.RULE:
-            response = await listEntities(
-              entityType,
-              alwaysBeExistingTopic,
-              alwaysBeExistingSubscription
-            );
-            break;
+      try {
+        await createEntity(
+          testCase.entityType,
+          testCase.alwaysBeExistingEntity,
+          testCase.parentTopicName,
+          testCase.parentSubscriptionName
+        );
+      } catch (err) {
+        log.httpAtomXml("Ignoring clean up step");
+      }
+      try {
+        await createEntity(
+          testCase.entityType,
+          testCase.alwaysBeDeletedEntity,
+          testCase.parentTopicName,
+          testCase.parentSubscriptionName
+        );
+      } catch (err) {
+        log.httpAtomXml("Ignoring clean up step");
+      }
 
-          default:
-            throw new Error("TestError: Unrecognized EntityType");
-        }
-        should.equal(Array.isArray(response), true, "Result must be any array for list requests");
-      });
+      response = await listEntities(
+        testCase.entityType,
+        testCase.parentTopicName,
+        testCase.parentSubscriptionName,
+        undefined,
+        1
+      );
 
-      it(`Updates an existent ${entityType} entity successfully`, async () => {
-        let response: any;
-        switch (entityType) {
-          case EntityType.QUEUE:
-            response = await updateEntity(entityType, alwaysBeExistingQueue);
-            should.equal(response.queueName, alwaysBeExistingQueue, "Queue name mismatch");
-            break;
-          case EntityType.TOPIC:
-            response = await updateEntity(entityType, alwaysBeExistingTopic);
-            should.equal(response.topicName, alwaysBeExistingTopic, "Topic name mismatch");
-            break;
-          case EntityType.SUBSCRIPTION:
-            response = await updateEntity(
-              entityType,
-              alwaysBeExistingSubscription,
-              alwaysBeExistingTopic
-            );
-            should.equal(
-              response.subscriptionName,
-              alwaysBeExistingSubscription,
-              "Subscription name mismatch"
-            );
-            break;
-          case EntityType.RULE:
-            response = await updateEntity(
-              entityType,
-              alwaysBeExistingRule,
-              alwaysBeExistingTopic,
-              alwaysBeExistingSubscription
-            );
-            should.equal(response.ruleName, alwaysBeExistingRule, "Rule name mismatch");
-            break;
-          default:
-            throw new Error("TestError: Unrecognized EntityType");
-        }
-      });
+      try {
+        await deleteEntity(
```

Comment:
It's starting to look like a lot of these tests depend on all entities being deleted before you start running them. If this is the case, I'd recommend using an `afterEach` or `beforeEach` block to delete all your entities. This ensures that a failure in one test doesn't leak state into another test (for example, if your `listEntities` call threw an error, then `deleteEntity` would never be called.

Reply:
Updated